src/Editor/EditorSupport.php

```diff
@@ -86,11 +66,19 @@ function () {
 	}
 
 	/**
-	 * Returns whether the editor in the current environment supports plugin features.
+	 * Returns whether the current environment is supported by the plugin
+	 * and the current screen is using the block editor.
 	 *
 	 * @return bool
 	 */
-	public function editor_supports_amp_block_editor_features() {
-		return $this->dependency_support->has_support();
+	public function supports_current_screen() {
```

Comment:
Maybe something like `is_current_screen_supported_block_editor_for_amp_enabled_post_type` would be more clear if also verbose. To me `supports_current_screen` is somewhat vague.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/6534/commits/8235b9f3700485964f46689cee21ea092846cd08.